site/src/sphinx/advanced-retry.rst

```diff
@@ -1,8 +1,154 @@
+
+    client.execute(...).aggregate().join();
+
+or even simply,
+
+.. code-block:: java
+
+    RetryStrategy strategy = RetryStrategy.onServerErrorStatus();
+    HttpClient client = new HttpClientBuilder(...)
+                          .decorator(RetryingHttpClient.newDecorator(strategy))
+                          .build();
+
+    client.execute(...).aggregate().join();
+
+That is it. The client will keep attempting until it succeeds or the number of attempts exceeds the default
+number of max attempts. Meanwhile, the ``strategy`` will decide to retry depending on the response.
+In this case, the client retries when it receives ``5xx`` response error.
+
+RetryStrategy
+-------------
+
+You can customize the ``strategy`` by implementing RetryStrategy_.
+
+.. code-block:: java
+
+    new RetryStrategy<HttpRequest, HttpResponse>() {
+        final Backoff backoff = RetryStrategy.defaultBackoff;
+
+        @Override
+        public CompletableFuture<Optional<Backoff>> shouldRetry(HttpRequest request,
+                                                                HttpResponse response) {
+            return response.aggregate().handle((result, cause) -> { // don't do get() or join()!
+                ...
+                if (result.headers().status() == CONFLICT) {
+                    return Optional.of(backoff);
+                }
+                return Optional.empty();
+            });
+        }
+    };
+
+This will retry when the response's status is ``409``.
+
+.. note::
+
+    We declare a ``backoff`` as a member and return it. The inside of RetryingClient_ tracks the
+    reference of the ``backoff`` and increases the counter one by one. The counter will be used
+    as the ``numAttemptsSoFar`` to get the next delay in Backoff_. we will take a close look what
+    the Backoff_ is at the bottom of this page.
+
+You can return different ``backoffs`` according to the response.
+
+.. code-block:: java
+
+    new RetryStrategy<HttpRequest, HttpResponse>() {
+        final Backoff backoffOnServerError = RetryStrategy.defaultBackoff;
+        final Backoff backoffOnConflict = Backoff.fixed(100);
+
+        @Override
+        public CompletableFuture<Optional<Backoff>> shouldRetry(HttpRequest request,
+                                                                HttpResponse response) {
+            return response.aggregate().handle((result, cause) -> {
+                ...
+                if (result.headers().status() == SERVER_ERROR) {
+                    return Optional.of(backoffOnServerError);
+                } else if (result.headers().status() == CONFLICT) {
+                      return Optional.of(backoffOnConflict);
+                }
+                return Optional.empty();
+            });
+        }
+    };
+
+.. _backoff-section:
+
+Backoff_
+--------
+
+You can use a ``backoff`` to determine the delay between attempts. There are three ``backoffs`` in Armeria:
+
+- FixedBackoff_
+- RandomBackoff_
+- ExponentialBackoff_
```

Comment:
I guess most users will not use these classes directly but only via the factory methods in `Backoff`. How about this:

- fixed delay, created with Backoff.fixed()
- random delay, created with Backoff.random()
- exponential delay which is multiplied on each attempt, created with Backoff.exponential()

Then we could perhaps simplify the next paragraph quite a bit and spend more space for the examples.